File: Level 1/개인정보 수집 유효기간/solution.js
function solution(today, terms, privacies) {
  const answer = [];

  const termMap = new Map();

  terms.forEach((term) => {
    const [type, month] = term.split(" ");
    termMap.set(type, +month);
  });

  privacies.forEach((privacy, index) => {
    const [date, type] = privacy.split(" ");
    const newDate = new Date(date);
    newDate.setMonth(newDate.getMonth() + termMap.get(type));

    if (newDate <= new Date(today)) {
      answer.push(index + 1);
    }
  });

  return answer;
}

File: Level 1/개인정보 수집 유효기간/solution.py
def to_days(date):
    year, month, day = map(int, date.split("."))
    return year * 12 * 28 + month * 28 + day


def solution(today, terms, privacies):
    term_map = {term[0]: int(term[2:]) * 28 for term in terms}
    today = to_days(today)

    answer = []

    for index, privacy in enumerate(privacies):
        date, term = privacy.split(" ")
        expire_date = to_days(date) + term_map[term]

        if expire_date <= today:
            answer.append(index + 1)

    return answer
